src/ReactiveSocket.cpp

```diff
@@ -203,40 +197,35 @@ bool ReactiveSocket::createResponder(
   switch (type) {
     case FrameType::SETUP: {
       Frame_SETUP frame;
-      if (frame.deserializeFrom(std::move(serializedFrame))) {
-        if (frame.header_.flags_ & FrameFlags_LEASE) {
-          // TODO(yschimke) We don't have the correct lease and wait logic above
-          // yet
-          LOG(WARNING) << "ignoring setup frame with lease";
-          //          connectionOutput_.onNext(
-          //              Frame_ERROR::badSetupFrame("leases not supported")
-          //                  .serializeOut());
-          //          disconnect();
-        }
-
-        auto streamState = handler->handleSetupPayload(ConnectionSetupPayload(
-            std::move(frame.metadataMimeType_),
-            std::move(frame.dataMimeType_),
-            std::move(frame.payload_),
-            false, // TODO: resumable flag should be received in SETUP frame
-            frame.token_));
-
-        connection.useStreamState(streamState);
-      } else {
-        // TODO(yschimke) enable this later after clients upgraded
-        LOG(WARNING) << "ignoring bad setup frame";
-        //        connectionOutput_.onNext(
-        //            Frame_ERROR::badSetupFrame("bad setup
-        //            frame").serializeOut());
-        //        disconnect();
+      if (!connection.deserializeFrameOrError(
+              frame, std::move(serializedFrame))) {
+        return;
       }
+      if (frame.header_.flags_ & FrameFlags_LEASE) {
+        // TODO(yschimke) We don't have the correct lease and wait logic above
+        // yet
+        LOG(WARNING) << "ignoring setup frame with lease";
```

Comment:
should we change this now?

The spec is pretty clear:

`If the server does NOT accept the contents of the SETUP frame, the server MUST send back a SETUP_ERROR and then close the connection.`

But reactivesocket-java used leases, so it was useful to ignore.